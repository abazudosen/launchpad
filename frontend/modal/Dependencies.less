@import './util.import.less';

.Dependencies {
  display: flex;
  flex-direction: column;
  align-items: stretch;

  &-Header {
    .font-s2;
    .type-bold;
    margin-bottom: 20px;
  }

  &-Text {
    .font-s1;

    code {
      background-color: rgba(245, 245, 245, 0.9);
      padding: 0 5px;
    }
  }

  &-List {
  }

  &-Item {
    .type-code;
    display: flex;
    flex-direction: row;
  }

  &-Name {
    width: 200px;
  }

  &-Version {
  }
}
