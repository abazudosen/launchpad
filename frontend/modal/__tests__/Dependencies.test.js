/* @flow */

import React from 'react';
import Dependencies from '../Dependencies';
import renderer from 'react-test-renderer';

describe('Dependencies Modal', () => {
  test('renders empty', () => {
    const component = renderer.create(<Dependencies dependencies={[]} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders non-empty', () => {
    const component = renderer.create(
      <Dependencies
        dependencies={[
          {
            name: 'graphql',
            version: '0.9.1S,',
          },
          {
            name: 'grahpql-tools',
            version: '1.0.0',
          },
        ]}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
