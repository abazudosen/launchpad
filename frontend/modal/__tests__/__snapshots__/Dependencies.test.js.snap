// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependencies Modal renders empty 1`] = `
<div
  className="Dependencies"
>
  <p>
    Dependencies are detected automatically from 
    <code>
      import
    </code>
     
    statements in your code when you save, and the latest versions are selected.
  </p>
</div>
`;

exports[`Dependencies Modal renders non-empty 1`] = `
<div
  className="Dependencies"
>
  <p>
    Dependencies are detected automatically from 
    <code>
      import
    </code>
     
    statements in your code when you save, and the latest versions are selected.
  </p>
  <div
    className="Dependencies-List"
  >
    <div
      className="Dependencies-Item"
    >
      <div
        className="Dependencies-Name"
      >
        graphql
      </div>
      <div
        className="Dependencies-Version"
      >
        0.9.1S,
      </div>
    </div>
    <div
      className="Dependencies-Item"
    >
      <div
        className="Dependencies-Name"
      >
        grahpql-tools
      </div>
      <div
        className="Dependencies-Version"
      >
        1.0.0
      </div>
    </div>
  </div>
</div>
`;
